### PR TITLE
tt support for pa

### PR DESCRIPTION
**Pull request by GlodBlock**
allow pa to use energy/laser hatches from tectech

### CODE REVIEW

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_ProcessingArray.java [hunk 1]
```diff
@@ -89,6 +90,38 @@ protected GT_Multiblock_Tooltip_Builder createTooltip() {
         return tt;
     }
 
+    @Override
```
Comment (repo-alt) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_ProcessingArray.java:93]: Perhaps it is not necessary to copy entire addToMachineList from the base class, just check for new energy hatches here?

Comment (GlodBlock) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_ProcessingArray.java:93]: yeah copy such big method is ugly, but PA uses the GT_MetaTileEntity_CubicMultiBlockBase as the super base, it only uses addToMachineList to add hatches.

Comment (Glease) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_ProcessingArray.java:93]: `return super.addToMachineList() || addNewEnergyHatches()`. Use super.xxx() can cut this clutter a lot. Same for the checkMachine.

Comment (GlodBlock) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_ProcessingArray.java:93]: ok, i will change it tomorrow. it's bed time for me.

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java [hunk 2]
```diff
@@ -939,6 +950,17 @@ public boolean addEnergyInputToMachineList(IGregTechTileEntity aTileEntity, int
         return false;
     }
 
+    public boolean addExoticEnergyInputToMachineList(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
+        if (aTileEntity == null) return false;
+        IMetaTileEntity aMetaTileEntity = aTileEntity.getMetaTileEntity();
+        if (aMetaTileEntity == null) return false;
+        if (aMetaTileEntity instanceof GT_MetaTileEntity_Hatch && GT_ExoticEnergyInputHelper.isExoticEnergyInput(aMetaTileEntity)) {
+            ((GT_MetaTileEntity_Hatch) aMetaTileEntity).updateTexture(aBaseCasingIndex);
+            return mExoticEnergyHatches.add((GT_MetaTileEntity_Hatch_Dynamo) aMetaTileEntity);
```
Comment (GlodBlock) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java:959]: dynamo :pog:

**Comment by draknyte1** [comment]
Multis also clear their hatch lists under certain conditions in onPostTick;
If you're not doing it in the base multi class, do it prior to calling super.onPostTick with the check being ==1.

Example:
```java
	@Override
	public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTick) {
		if (aBaseMetaTileEntity.isServerSide()) {
			if (mUpdate == 1 || mStartUpCheck == 1) {
				this.mTurbineRotorHatches.clear();
			}
		}
		super.onPostTick(aBaseMetaTileEntity, aTick);
	}
```

**Review by repo-alt — APPROVED** [review]

**Comment by repo-alt** [comment]
The list probably should be cleared in the base, where it belongs

**Pull request merged**